### ISSUE TITLE
Multi-Speaker TTS with Speaker Encoder and GST

### ISSUE THREAD

**Issue by erogol** [mozilla/TTS#485]
Below are Colabs notebooks for Demo of two models, the demos allow upload of GST samples, upload of wav file to use as a reference for the speaker (synthesize with your own voice).

[Demo Mozilla TTS MultiSpeaker Jia et al. 2018](https://colab.research.google.com/drive/1OX2jtyxmeFt7kRO65QgPy65ku6HWUrFK?usp=sharing)
[Demo Mozilla TTS MultiSpeaker Jia et al. 2018 With GST](https://colab.research.google.com/drive/1JX29gDcsPHx5HlbwNkpHnPQkhhrSvrji?usp=sharing)

Both models were trained using the VCTK dataset. All checkpoints, configs and logs are available on the Google Drive:
[Mozilla TTS MultiSpeaker Jia et al. 2018](https://drive.google.com/drive/folders/15R0XOs5iTckNIaP7pKpYaTFwxW4rT1CD?usp=sharing)
[Mozilla TTS MultiSpeaker Jia et al. 201 With GST](https://drive.google.com/drive/folders/15HE-5dfPi7AwVta0FfUimNOwaOLifNVX?usp=sharing)

The models synthesize well, but when changing the sample of extraction of the embeddings the synthesized voice is not very close to the reference voice. Perhaps the problem is the Speaker Encoder. I think we should train with more data and try to use the Angular Prototypical that achieved state of the art results in  the paper ["In defence of metric learning for speaker recognition"](https://arxiv.org/abs/2003.11982)

_Originally posted by @Edresson in https://github.com/mozilla/TTS/pull/472#issuecomment-666606538_

**Comment by george-roussos** [comment]
My tests highlight the importance of many speakers. I needed to have at least 150 different voices and 30 minutes of speech from each to get good generalization. Also, I used CorentinJ's speaker encoder to get the embeddings and I restored on a single speaker model. With 150 speakers (of one gender) generalization is very good, but not perfect. I would guess for really good results we would need somewhere around 500 speakers.
